libs/public-pages/src/file-storage/file-retrieval.ts

```diff
@@ -0,0 +1,65 @@
+import fs from "node:fs/promises";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import { getContentTypeFromExtension } from "./content-type.js";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+// Navigate to monorepo root (from libs/public-pages/src/file-storage/)
+const MONOREPO_ROOT = path.join(__dirname, "..", "..", "..", "..");
+const STORAGE_BASE = path.join(MONOREPO_ROOT, "storage", "temp", "uploads");
+
+async function findFileByArtefactId(artefactId: string): Promise<{ buffer: Buffer; extension: string } | null> {
```

Comment:
I wonder if this sits better in the publication lib, as this is going to be used in many places in the future like subscriptions fulfilment, deletion of artefacts etc